lib/validator/validator.go

```diff
@@ -301,6 +302,21 @@ func validatePoolMetadata(v *validator.Validate, structLevel *validator.StructLe
 	}
 }
 
```

Comment:
You'd need to do this by adding a validator for the IPPool since you'd need to validate across the Metadata and the Spec.